controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -920,23 +920,25 @@ bool tlvPrefilledData::add_tlv_value_hex_string(const std::string &value, uint16
  */
 bool tlvPrefilledData::add_tlv_value_decimal_string(const std::string &value, uint16_t &length)
 {
-    int i_dec           = std::stoi(value.substr(2, value.length()));
-    size_t num_of_bytes = 1;
-    // Assume 2 bytes if it has more than 3 digits
-    if (value.size() > 5) {
-        num_of_bytes++;
+    /*
+    We don't really know how to identify if it is one or two (or more) bytes.
+    If you encounter this problem now, go back to the source code as a hint to
+    what is going on.
+    */
+
+    // Convert to hex string
+    std::stringstream ss;
+    ss << std::hex << std::stoi(value.substr(2, value.length()));
```

Comment:
Again: we *know* this is wrong, because there are cases where there's a 0d with leading zeroes. For example, in the test specification 5.8.3, we have:

> Trigger (DEV_SEND_1905, DestALid,
> WTS_REPLACE_CTTAGENT1_ALID,
> MessageTypeValue,0x8016,
> tlv_type,0x9D,tlv_length,0xXXXX,tlv_value,{
> WTS_REPLACE_CTTAGENT1_BSSID
> 0x00 ***0d0030*** 0x01
> {WTS_REPLACE_CTT_STA_MAC} })
>
> MCUT to send Client Association Control
> Request message to CTT Agent1 with:
>
> - [...]
> - Validity Period field set to 30 seconds
> - [...]

(my emphasis). The validity period `0d0030` has to be two bytes in the TLV, but the above conversion to string will turn it into `0x1e`.

So, if you do want to convert it to string, it has to be:

```
    size_t num_of_bytes = 1;
    // We don't really know how to identify if it is one or two (or more) bytes.
    // Assume 2 bytes if it has more than 3 digits
    if (value.size() > 5) {
        num_of_bytes++;
    }
    ss << "0x" << std::setw(num_of_bytes * 2) << std::setfill('0') << std::hex 
       << std::stoi(value.substr(2, value.length()));
```

(This includes my earlier suggestion of including the additional formatting in the strstream output. So all the code below can be removed.)